pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -35,6 +36,61 @@ makeWrapper() {
 
     assertExecutable "$original"
 
+    # Write wrapper code which adds `value` to the beginning or end of
+    # the list variable named by `varName`, depending on the `mode`
+    # specified.
+    #
+    # A value which is already part of the list will not be added
+    # again. If this is the case and the `suffix` mode is used, the
+    # list won't be touched at all. The `prefix` mode will however
+    # move the first matching instance of the value to the beginning
```

Comment:
```suggestion
    # move one matching instance of the value to the beginning
```
(Extreme pedantry: the regex does not always pick the first value. I don't claim to know why.)

Reply:
Oh yeah, this will always match the _last_ instance of the value, since `.*` will always greedily match as much as it can. AFAIK there's no way to make it lazy, but it shouldn't really matter which instance is moved anyway.

Reply:
It does catch the first entry, maybe when the new value is at the front. But it's not important.

Reply:
That's strange. It shouldn't and doesn't do that in my tests. Do you have an example?

Reply:
```bash
$ [[ "a:b:b:c:b:d:f" =~ ^((.*):)?"b"(:(.*))?$ ]] && echo "${BASH_REMATCH[2]:+:${BASH_REMATCH[2]}}${BASH_REMATCH[4]:+:${BASH_REMATCH[4]}}"
:a:b:b:c:d:f
```
For whatever reason, this removes the third `b`.

Reply:
Yeah, the third `b` is the last instance, so that's correct.

Reply:
Silly me, you were asking for an example about the first entry. Indeed, I can't reproduce it. I was probably playing with slightly different REs when I saw the different behaviour. You can mark as resolved.